src/policy/immix/immixspace.rs

```diff
@@ -23,21 +23,26 @@ use crate::vm::*;
 use crate::{
     plan::ObjectQueue,
     scheduler::{GCWork, GCWorkScheduler, GCWorker, WorkBucketStage},
-    util::{
-        heap::FreeListPageResource,
-        opaque_pointer::{VMThread, VMWorkerThread},
-    },
+    util::opaque_pointer::{VMThread, VMWorkerThread},
     MMTK,
 };
 use atomic::Ordering;
 use std::sync::{atomic::AtomicU8, Arc};
 
+// For 32-bit platforms, we still use FreeListPageResource
+#[cfg(target_pointer_width = "32")]
+use crate::util::heap::FreeListPageResource as ImmixPageResource;
+
+// BlockPageResource is for 64-bit platforms only
```

Comment:
This does not really depend on architecture. It depends on the space being contiguous or discontiguous. However, that is not a compile-time flag. I am okay with checking pointer width, but you should make it clear that this is a hack.

Alternatively, see my comments on the `BlockPageResource`. If we can support discontiguous page resource for `BlockPageResource`, we will no longer need this conditional compilation.

Reply:
Fixed. The conditional compilation is removed.